networklibrary/src/main/java/com/eis/communication/network/SMSNetDictionary.java

```diff
@@ -0,0 +1,55 @@
+package com.eis.communication.network;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Concrete implementation of a INetworkDictionary
+ *
+ * @author Marco Cognolato
+ */
+public class SMSNetDictionary implements INetworkDictionary<String, String> {
```

Comment:
Are you restricting to `String` only for now? Did you take account that one day we could be able to handle whatever type of `Serializable` object? Would you implement the serialization and de-serialization at dictionary level or at manager level?
Not saying that you should to it now, not at all, just asking if you already thought about this and if you needed any kind of help.

Reply:
String is only for now, yes.
About serialization I'll probably either do it at manager level, or use the Strategy Pattern to let the user do it, I don't have an exact plan as of today. So yes, any help is greatly appreciated 👍